python/foglamp/services/core/api/filters.py

```diff
@@ -240,50 +282,227 @@
+        cf_mgr = ConfigurationManager(storage)
+
+        # Fetch the filter items: get category items
+        category_info = await cf_mgr.get_category_all_items(category_name=service_name)
+        if category_info is None:
+            # Error service__name doesn't exist
+            message = "No such '%s' category found." % service_name
+            return web.HTTPNotFound(reason=message)
+
+        filter_value_from_storage = json.loads(category_info['filter']['value'])
+    except KeyError as ex:
+        _LOGGER.exception("Get pipeline: {}, no filter exists: ".format(service_name) + str(ex))
+        raise web.HTTPNotFound(reason="Get pipeline: {}, no filter exists: ".format(service_name) + str(ex))
+    except StorageServerError as ex:
+        _LOGGER.exception("Get pipeline: {}, caught exception: ".format(service_name) + str(ex))
+        raise web.HTTPInternalServerError(reason=str(ex))
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'result': filter_value_from_storage})
+
+
+async def delete_filter(request):
+    """ DELETE filter
+
+    :Example:
+        curl -X DELETE http://localhost:8081/foglamp/filter/<filter_name>
+    """
+    filter_name = request.match_info.get('filter_name', None)
+    if filter_name is None:
+        raise web.HTTPNotFound(reason="Filter name is required.")
+    try:
+        storage = connect.get_storage_async()
+
+        # Check if it is a valid plugin
+        payload = PayloadBuilder().WHERE(['name', '=', filter_name]).payload()
+        result = await storage.query_tbl_with_payload("filters", payload)
+        if len(result["rows"]) == 0:
+            raise web.HTTPNotFound(reason="No such filter '{}' found".format(filter_name))
+
+        # Check if filter exists in any pipeline
+        payload = PayloadBuilder().WHERE(['name', '=', filter_name]).payload()
+        result = await storage.query_tbl_with_payload("filter_users", payload)
+        if len(result["rows"]) != 0:
+            raise web.HTTPBadRequest(reason="Filter '{}' found in pipelines".format(filter_name))
+
+        # Delete filter from filters table
+        payload = PayloadBuilder().WHERE(['name', '=', filter_name]).payload()
+        await storage.delete_from_tbl("filters", payload)
+
+        # Delete configuration for filter
+        _delete_configuration_category(storage, filter_name)
+    except StorageServerError as ex:
+        _LOGGER.exception("Get filter: {}, caught exception: ".format(filter_name) + str(ex))
+        raise web.HTTPInternalServerError(reason=str(ex))
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'result': "Filter {} deleted successfully".format(filter_name)})
+
+
+async def delete_filter_pipeline(request):
+    """ DELETE filter pipeline
+
+    :Example:
+        curl -X DELETE http://localhost:8081/foglamp/filter/<service_name>/pipeline
```

Comment:
Not sure about this route, if idea is to set empty pipeline, we can do it via `curl -X PUT -d '{"pipeline": []}' http://localhost:8081/foglamp/filter/service_name/pipeline `